packages/@aws-cdk/aws-codepipeline-actions/lib/s3/source-action.ts

```diff
@@ -110,7 +110,12 @@ export class S3SourceAction extends Action {
   protected bound(_scope: Construct, stage: codepipeline.IStage, options: codepipeline.ActionBindOptions):
   codepipeline.ActionConfig {
     if (this.props.trigger === S3Trigger.EVENTS) {
-      const id = stage.pipeline.node.uniqueId + 'SourceEventRule' + this.props.bucketKey;
+      const id = stage.pipeline.node.uniqueId + 'SourceEventRule' +
+        // we can't use Tokens in construct IDs,
+        // however, if bucketKey is not a Token,
+        // we want it to differentiate between multiple actions
+        // observing the same Bucket with different keys
+        (Token.isUnresolved(this.props.bucketKey) ? '' : this.props.bucketKey);
```

Comment:
Using an empty string here is good enough for now, but I can already taste the next bug report coming in: "hey it doesn't work if I use 2 S3 source triggers based off of tokens"